solver/errdefs/errdefs.proto

```diff
@@ -0,0 +1,42 @@
+syntax = "proto3";
+
+package errdefs;
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+
+option (gogoproto.stable_marshaler_all) = true;
+
+message Stack {
+    repeated Frame frames = 1;
+    repeated string Cmdline = 2;
+    string Checksum = 3;
+    int32 Pid = 4;
+    // go commit, frontend digest
```

Comment:
Is the intention here to eventually support propagating `errors.WithStack` from within frontends?

Reply:
Yes. You can look at https://gist.github.com/tonistiigi/3f3cfcf274f5dde5b7cb69f44f44ed84 for an example. That trace contains stacks from buildctl, buildkitd and two dockerfile containers (one that does the build and other that forwards #syntax). Bit verbose because of https://github.com/pkg/errors/issues/75 but we're not going to show this part to users by default anyway.

Reply:
Awesome, this is really useful.